watcher/lib/influxdb/rollups.go

```diff
@@ -145,10 +96,11 @@ func buildFunction(f Function) (string, error) {
 	}
 
 	// split function name, based on the "_" separator (eg: percentile_99, top_10, ecc)
-	funcAndValue := strings.Split(f.Function, "_")
-	funcName := funcAndValue[0]
-	if len(funcAndValue) == 2 && isFuncWithParams(funcName) {
+	if isCompositeFunc(f) {
+		funcAndValue := strings.Split(f.Function, "_")
+		funcName := funcAndValue[0]
 		param := funcAndValue[1]
```

Comment:
You still don't check that `funcAndValue` is a 2-element slice and thus this line can cause "index out of range" panic. You can add a length check after you split the functions:

```go
funcAndValue := strings.Split(f.Function, "_")
if len(funcAndValue) != 2 {
  return "", trace.BadParameter(`incorrectly formatted composite function %q, expected "<name>_<parameter>"`,
    f.Function)
}
```

Reply:
I added additional checks in the Check() function, so that it's an early detection and it seemed cleaner in there.

Reply:
That's fine to have it in the `Check()` method, but it does not address the fact that this line is a potential panic. Imagine someone calls this method without calling `Check()` first.

Reply:
gotcha... I guess :) checkout the latest commit